pytezos-tests/pytezos_contract_tester.py

```diff
@@ -327,14 +327,22 @@ def __init__(self):
         self.fees = []
     
     def add_result(self, op_name, result):
-        """Add the fees from the operation result to self.fees"""
-        fee = int(result['contents'][0]['fee'])
-        storage_bytes = int(result['contents'][0]['storage_limit'])
-        storage_cost = int(storage_bytes) * 250 # 250 mutez per storage_bytes byte on edo
-        gas = int(result["contents"][0]["metadata"]["operation_result"]["consumed_gas"])
-        total_cost = fee + storage_cost
-        fee = {"total_cost":total_cost, "fee":fee, "storage_bytes":storage_bytes, "storage_cost":storage_cost, "gas":gas}
-        self.fees.append({op_name:fee})
+        """Add the fees from the operation costs to self.fees"""
+        costs = {}
+        op_metadata = result["contents"][0]["metadata"]["operation_result"]
+        costs["gas"] = int(op_metadata["consumed_gas"])
+        costs["fee"] = int(result['contents'][0]['fee'])
+        storage_bytes = 0
+        if "paid_storage_size_diff" in op_metadata:
+            storage_bytes = int(op_metadata["paid_storage_size_diff"])
+        costs["storage_bytes"] = storage_bytes
+        costs["storage_cost"] = int(storage_bytes) * 250 # 250 mutez per storage_bytes byte on edo
+        costs["total_cost"] = costs["fee"] + costs["storage_cost"]
+        # "originate" operation incurs a fixed allocation_fee for creating a new contract address
+        if op_name == "originate":
+            costs["allocation_fee"] = 64250
```

Comment:
Is this 250 mutez per storage bytes fixed for just Edo or is it different for each protocol? Also, the same question for the allocation fee. 

If they are different for each Tezos protocol, then we should definitely make these more configurable so that we don't have to manually update the script.

Reply:
Yes that's a good point. It's a protocol constant, so it is subject to change.
I added a small get request to get the latest protocol constants from main net (using tzstats) for saving the costs.